src/cljc/matcher_combinators/matchers.cljc

```diff
@@ -31,6 +31,19 @@
         :else
         (core/->Value expected)))
 
+(declare match-with)
+
+(defn strictly-equals
+  "Return a matcher for the expected value which matches nested maps with `equals`.
+  This solves a common need of matching nested maps with strict equality.
+  See also: `match-with`.
+
+  Usage:
+    (is (match? (strictly-equals {:a :b})
+                {:a :b :c :d}))"
```

Comment:
this example isn't a case where using `strictly-equals` would be any different than from using `equals` at the top level.
I'd go with something that shows the difference from

```clojure
(is (match? (m/equals {:user1 {:id 5}
                       :user2 {:id 3}})
            {:user1 {:id 5 :name "hennix"}
             :user2 {:id 3 :name "flynt"}}))
```
which gives different results than
```clojure
(is (match? (m/equals {:user1 (m/equals {:id 5})
                       :user2 (m/equals {:id 3})})
            {:user1 {:id 5 :name "hennix"}
             :user2 {:id 3 :name "flynt"}}))
```

Reply:
Yeah, I was thinking to put this case just to inform it maintain the `equals` behavior (similar to using the `match-with`), but I will change it to show the cases where it would be helpful